code/modules/vore/eating/bellymodes_vr.dm

```diff
@@ -472,5 +472,8 @@
 			put_in_egg(P,1)
 
 	if(play_sound)
+		for (var/mob/living/M in contents)
+			if(M.client.is_preference_enabled(/datum/client_preference/digestion_noises))
+				M << sound(play_sound)
 		playsound(src, play_sound, vol = 100, vary = 1, falloff = VORE_SOUND_FALLOFF, ignore_walls = FALSE, preference = /datum/client_preference/digestion_noises)
```

Comment:
Okay I figured it out.  It is because of ye old issue where hearers() will not include people inside bellies.   It can be fixed by changing this line to `ignore_walls = TRUE`  then you won't need the loop.

Reply:
Good find

Reply:
Doesn't explain the playsound_local not working for them either tho.

Reply:
https://github.com/VOREStation/VOREStation/commit/2fc033ee0eeb76a2665465403c8c807871ee439a
Yeah I'm pretty sure the get_step thing here draws from the thing's xyz coordinates while the non-turf loc getting substituted with the working locs list feature was removed

Reply:
`get_turf()` works perfectly fine and is faster than that old method.   get_step() is the method used by all codebases for years.   If you don't believe me, try it.

Reply:
also you may want to do `if(M.client && M.client.is_preference_enabled` just to ensure that byond actually knows there's a client there. it's an annoying thing.

Reply:
Well I'd provide this exact thing here with the deaf prey thing. The prey's coordinates inside the modern obj based belly is 0,0,0, and the playsound_local shrinks the volume to none from the distance from there to the sound source turf or at least that's what I think is happening. The ignore_walls part is only present at the playsound, and not in the playsound_local, whic has been tested completely dead for the prey as well.

Reply:
so just call `M.playsound_local(get_turf(owner), play_sound, 100)` 

because the belly is inside of the predator, who owns it. and the prey are inside of the obj/belly, so if you get the owner of said belly's turf, you're fine. It's how I did it and y'all have the same sound code as citadel does.

Reply:
Calling `get_turf(prey)` where prey is inside a belly will return the /turf that the pred is standing on.

Reply:
Alright gonna test that when I have time. There's still the problem that the playsound calculations both are based also on the prey mob's get_turf hardcoded into the playsound_local as calling the get_turf(src) for the receiving mob.

Reply:
Alright gonna try the stuff when I get the chance. I still do kinda think that the ignore_walls thing was put there because of people getting annoyed by things carrying over from neighboring dorms to others and such.

Reply:
You were right. Thanks